### ISSUE TITLE
Fix unwanted "picked*" events from pinch-to-zoom on touch devices

### ISSUE THREAD

**Issue by juho-ylikyla** [xeokit/xeokit-sdk#438]
**Describe the bug**

On touch devices, when doing the pinch to zoom gesture (dollying), MousePickHandler sometimes triggers "picked" events.

**To Reproduce**
1. Go to any example project hosted on xeokit on a touch device
2. Do a couple of pinch to zooms
3. See events triggered

**Problem**
As far as I understand, this should prevent pick events when the touch/press down and up events are "too close" to each other

https://github.com/xeokit/xeokit-sdk/blob/c761f7073ee9da57bceea2d55006a4a7eed5938b/src/viewer/scene/CameraControl/lib/handlers/MousePickHandler.js#L184

`mouseDownClientX/Y` are however never updated correctly on touch devices.

**Possible fix**

I fixed this locally by removing these two lines...

https://github.com/xeokit/xeokit-sdk/blob/c761f7073ee9da57bceea2d55006a4a7eed5938b/src/viewer/scene/CameraControl/lib/handlers/MousePickHandler.js#L135-L136

...and adding the functionality after this line

https://github.com/xeokit/xeokit-sdk/blob/c761f7073ee9da57bceea2d55006a4a7eed5938b/src/viewer/scene/CameraControl/lib/handlers/TouchPanRotateAndDollyHandler.js#L47

``` if (touches.length === 1 && changedTouches.length === 1) {
                tapStartTime = states.touchStartTime;
                tapStartPos[0] = touches[0].pageX;
                tapStartPos[1] = touches[0].pageY;
    
                states.mouseDownClientX = touches[0].clientX;
                states.mouseDownClientY = touches[0].clientY;
```

Note that this will most likely break the picked event on non-touch devices.

**Comment by xeolabs** [comment]
Thanks @juho-ylikyla, which OS/device are you using?

**Comment by xeolabs** [comment]
@juho-ylikyla thanks, that gave good clues. The issue is caused by mouse handlers getting events that touch handlers should have consumed, as described [here](https://developer.mozilla.org/en-US/docs/Web/API/Touch_events/Supporting_both_TouchEvent_and_MouseEvent).

Fix committed in 22aad46
